machinelearning/tensorflow.md

```diff
@@ -0,0 +1,405 @@
+# ML inference on Graviton CPUs with TensorFlow
+
+**Introduction**
+
+TensorFlow is an open-source software library for machine learning and artificial intelligence. It can be used across training and inference of deep neural networks. This document covers how to use Tensorflow based machine learning inference on Graviton CPUs, what runtime configurations are important and how to debug any performance issues. The document also covers instructions for source builds and how to enable some of the downstream features.
+
+# How to use Tensorflow on Graviton CPUs
+To start with how to use Tensorflow inference on Graviton CPUs, there are multiple levels of software package abstractions available: Python wheel (quick option), Docker container (comes with wheel, additional packages and benchmarks) and DLAMI (Deep Learning Amazon Machine Image that comes with preinstalled packages and tools). Let's see how one can use each of them to get started.
+
+**Using Python wheel**
+Tensorflow official supports optimized onednn+acl backend for Graviton CPUs.
+```
+pip install tensorflow-cpu-aws
+```
+
+**Using Docker hub container**
+
+Launch a Graviton instance with an Ubuntu 20.04 AMI and install docker using the below instructions.
```

Comment:
Ubuntu 20.04 is not a requirement as docker images work on any distro that has docker installed.  Instead, remove the docker installation instructions, or link to another howto for Ubuntu, AL2 etc.  Simply include the the steps to download the image and run it.

Reply:
ok